activerecord/lib/active_record/connection_handling.rb

```diff
@@ -45,16 +45,20 @@ module ConnectionHandling
     # The exceptions AdapterNotSpecified, AdapterNotFound and +ArgumentError+
     # may be returned on an error.
     def establish_connection(spec = nil)
+      raise RuntimeError, "Anonymous class is not allowed." unless name
+
       spec     ||= DEFAULT_ENV.call.to_sym
       resolver =   ConnectionAdapters::ConnectionSpecification::Resolver.new configurations
-      spec     =   resolver.spec(spec)
+      # TODO: uses name on establish_connection, for backwards compatibility
+      spec     =   resolver.spec(spec, self == Base ? "primary" : name)
```

Comment:
Nah, after a quick dip in irb it seems `==` will do fine here. Got confused because I've only ever seen class comparisons done with `===` so thought something was a foot (but everything was in hand 🤓).